File: solutions/kotlin/difference-of-squares/1/src/main/kotlin/DifferenceOfSquares.kt
class Squares(count: Int) {
    private val cnt = count

    fun sumOfSquares(): Int {
        return (0..cnt).sumOf { it * it }
    }

    fun squareOfSum(): Int {
        return (0..cnt).sum().let { it * it }
    }

    fun difference(): Int {
        return squareOfSum() - sumOfSquares()
    }
}

File: solutions/scala/difference-of-squares/1/src/main/scala/DifferenceOfSquares.scala
object DifferenceOfSquares {

  def sumOfSquares(n: Int): Int = (1 to n).map(x => x * x).sum

  def squareOfSum(n: Int): Int = Math.pow((1 to n).sum, 2).toInt

  def differenceOfSquares(n: Int): Int = (sumOfSquares(n) - squareOfSum(n)).abs
}
